omnisafe/common/buffer/vector_onpolicy_buffer.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2022 OmniSafe Team. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implementation of vector on-policy buffer."""
+
+from typing import Dict
+
+import torch
+
+from omnisafe.common.buffer.onpolicy_buffer import OnPolicyBuffer
+from omnisafe.typing import AdvatageEstimator, OmnisafeSpace
+from omnisafe.utils import distributed_utils
+
+
+class VectorOnPolicyBuffer(OnPolicyBuffer):
+    """Vectorized on-policy buffer."""
+
+    def __init__(  # pylint: disable=super-init-not-called, too-many-arguments
+        self,
+        obs_space: OmnisafeSpace,
+        act_space: OmnisafeSpace,
+        size: int,
+        gamma: float,
+        lam: float,
+        lam_c: float,
+        advantage_estimator: AdvatageEstimator,
+        penalty_coefficient: float,
+        standardized_adv_r: bool,
+        standardized_adv_c: bool,
+        num_envs: int = 1,
+        device: torch.device = torch.device('cpu'),
+    ):
```

Comment:
docstring

Reply:
docstring already defined in the BaseBuffer(abstract class)